src/frontend/src/views/Upload/index.tsx

```diff
@@ -35,6 +35,11 @@ export default function Upload(): JSX.Element | null {
   const pathogen = useSelector(selectCurrentPathogen);
   const [samples, setSamples] = useState<ISamples | null>(null);
   const [metadata, setMetadata] = useState<SampleIdToMetadata | null>(null);
+  const [hasManuallyEditedMetadata, setHasManuallyEditedMetadata] =
```

Comment:
Do all three of the upload components really need to know this state? It seems to me it's likely only needed on the metadata step.

Reply:
no but i didn't see another way to let the review page know how the metadata had been edited from the metadata page, do you know of a better way?

Reply:
I think for the Review step to have access to the information of how the metadata was set so the analytics event can include it, this is probably the most reasonable way.

Reply:
yep

Reply:
I don't think sending this on the review step is necessarily the right call. What are we trying to track specifically? Any time a person does something metadata related? Their first upload/edit? The last? 

IMO, if it's about tracking actions on the metadata page, then the state should live in that component only, the events should fire from that component, and the Review component should not be involved.

If necessary, let's acquire a clearer understanding of what/when we need to track.

Reply:
so we want to track how the user is entering metadata, whether it's via manual entry or tsv upload. the issue with doing this on the metadata page is that it's hard to tell when the user is done with editing metadata. We don't want to trigger events multiple times if they enter data via tsv and then use the webform. I added this event to the review step because by that time the user would have been done entering metadata and we can trigger only one analytics event.

Reply:
@mmmmmmaya is that explanation enough? Can we merge this?

Reply:
Ok, I thought about this a bit and I think we can merge it as-is for now. However, I would like you to make a ticket for the eng backlog and post it on this thread and in a slack channel cc Christine that we should refactor the upload components. I think a couple of things here have code smell to me. I don't like the idea that all 3 need to have the same props with the way things are set up, especially if some of those props will not be used, and I think the state should probably be managed in `Upload` instead of the metadata page if it's used by multiple children.

Reply:
https://app.shortcut.com/genepi/story/229448/refactor-upload-flow-shared-props